pkg/api/datasources.go

```diff
@@ -126,6 +128,12 @@ func DeleteDataSourceByName(c *models.ReqContext) Response {
 
 func AddDataSource(c *models.ReqContext, cmd models.AddDataSourceCommand) Response {
 	cmd.OrgId = c.OrgId
+	if cmd.Url != "" {
+		_, err := url.Parse(cmd.Url)
+		if err != nil {
+			return Error(400, "Validation error", errutil.Wrapf(err, "invalid data source URL %q", cmd.Url))
```

Comment:
Message in UI is not super friendly
![image](https://user-images.githubusercontent.com/1668778/79860159-7e168f80-83d2-11ea-8547-2d7cbe5f5143.png)

Should say something along the lines "Invalid URL".

Reply:
@marefr I think this must be fixed in the UI, because that's not the message sent from the API (I checked in the browser). Shall we fix it in another PR?

Reply:
@marefr I think I see how to fix it.